java/code/src/com/suse/scc/SCCSystemRegistrationManager.java

```diff
@@ -58,6 +64,41 @@ public SCCSystemRegistrationManager(SCCClient sccClientIn) {
         this.sccClient = sccClientIn;
     }
 
+    /**
+     * Update last_seen field in SCC for all registered clients
+     */
+    public void updateLastSeen() {
+        List<Credentials> credentials = CredentialsFactory.lookupSCCCredentials();
+        Credentials primCred = credentials.stream()
+                .filter(c -> c.isPrimarySCCCredential())
+                .findFirst().orElseThrow();
+        List<Map<String, Object>> candidates = SCCCachingFactory.listUpdateLastSeenCandidates(primCred);
+
+        List<SCCUpdateSystemJson> sysList = candidates.stream()
+                .map(c -> new SCCUpdateSystemJson(
+                        c.get("scc_login").toString(),
+                        c.get("scc_passwd").toString(),
+                        (Date) c.get("checkin")))
+                .collect(Collectors.toList());
+
+        ArrayList<List<SCCUpdateSystemJson>> batches = new ArrayList<>(
+                IntStream.range(0, sysList.size()).boxed().collect(
+                        Collectors.groupingBy(e -> e / Config.get().getInt(ConfigDefaults.REG_BATCH_SIZE, 199),
+                                Collectors.mapping(e->sysList.get(e), Collectors.toList())
+                                )).values());
+        for (List<SCCUpdateSystemJson> batch: batches) {
+            try {
+                sccClient.updateBulkLastSeen(batch, primCred.getUsername(), primCred.getPassword());
+            }
+            catch (SCCClientException e) {
+                LOG.error("SCC error while updating systems", e);
```

Comment:
What are the pros and cons of using `setRegistrationErrorTime` here?

Reply:
We do not know which client cause the error. We would need to block "all in the batch" for a 1 week.
Even if it was a problem on SCC side.
As this bulk update happens only 1 time per day, I think it is ok to do it again in ~24 hours.
In case a single client update create a problem, this client is excluded from the batch.